src/aap_eda/services/ruleset/activation_kubernetes.py

```diff
@@ -288,27 +289,35 @@ def run_activation_job(
         namespace,
         activation_instance,
         secret_name,
+        reconnect,
     ) -> None:
-        # wait until old job instance has completely shut down.
-        done = False
-        while not done:
-            if_job_exists = self.batch_api.list_namespaced_job(
-                namespace=namespace,
-                label_selector=f"job-name={job_name}",
-                timeout_seconds=0,
+        if reconnect:
+            job_list = self.batch_api.list_namespaced_job(
+                namespace=namespace, label_selector=f"job-name={job_name}"
             )
+            if not job_list.items:
+                raise ReconnectContainerNotFound(f"Job {job_name} not found")
+        else:
+            # wait until old job instance has completely shut down.
+            done = False
+            while not done:
+                if_job_exists = self.batch_api.list_namespaced_job(
+                    namespace=namespace,
+                    label_selector=f"job-name={job_name}",
+                    timeout_seconds=0,
+                )
 
-            if not if_job_exists.items:
-                break
+                if not if_job_exists.items:
```

Comment:
If this always return `True`, we will stuck here. can we force it to break after sleep a while?

Reply:
This is the existing logic (when reconnect == False). We should fix it in another PR if it is a real bug.